File: src/test/java/io/flexwork/service/SignupServiceTest.java
package io.flexwork.service;

import static io.flexwork.stateMacine.signup.SignupStates.NEW_SIGNUP_USER;
import static org.assertj.core.api.Assertions.assertThat;

import io.flexwork.IntegrationTest;
import io.flexwork.domain.User;
import io.flexwork.stateMacine.signup.SignupEvents;
import io.flexwork.stateMacine.signup.SignupStates;
import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.statemachine.StateMachine;
import org.springframework.statemachine.config.StateMachineFactory;
import org.springframework.statemachine.state.State;

@IntegrationTest
public class SignupServiceTest {

    @Autowired
    private SignupService signupService;

    @Autowired
    private StateMachineFactory<SignupStates, SignupEvents> stateMachineFactory;

    @Test
    void signup() {
        User user = new User();
        user.setId("123");
        user.setLogin("hainguyenLogin");
        user.setSignupState(NEW_SIGNUP_USER);
        user.setEmail("test@test.com");
        signupService.signup(user);

        assertThat(user.getSignupState()).isEqualTo(NEW_SIGNUP_USER);
        StateMachine<SignupStates, SignupEvents> state = stateMachineFactory.getStateMachine("111");
        System.out.println("He " + state);
    }
}

File: src/test/kotlin/io/flexwork/service/SignupServiceTest.kt
package io.flexwork.service

import io.flexwork.IntegrationTest
import io.flexwork.domain.User
import io.flexwork.stateMacine.signup.SignupEvents
import io.flexwork.stateMacine.signup.SignupStates
import org.assertj.core.api.Assertions
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.statemachine.service.StateMachineService
import kotlin.test.Test

@IntegrationTest
class SignupServiceTest {
    
    @Autowired
    private val signupService: SignupService? = null

    @Autowired
    private val stateMachineService: StateMachineService<SignupStates, SignupEvents>? = null

    @Test
    fun signup() {
        val user = User()
        user.id = "123"
        user.login = "hainguyenLogin"
        user.signupState = SignupStates.NEW_SIGNUP_USER
        user.email = "test@test.com"
        signupService!!.signup(user)

        Assertions.assertThat(user.signupState).isEqualTo(SignupStates.NEW_SIGNUP_USER)
        val state = stateMachineService!!.acquireStateMachine("signu11233p-" + user.id)
        println("He " + state.state)
    }
}
